src/engine/renderer/Material.cpp

```diff
@@ -989,40 +995,190 @@ void MaterialSystem::GenerateWorldMaterialsBuffer() {
 void MaterialSystem::GenerateWorldCommandBuffer() {
 	Log::Debug( "Generating world command buffer" );
 
-	uint32_t count = 0;
-	for ( const MaterialPack& pack : materialPacks ) {
-		for ( const Material& material : pack.materials ) {
-			count += material.drawCommands.size();
+	totalBatchCount = 0;
+
+	uint32_t batchOffset = 0;
+	uint32_t globalID = 0;
+	for ( MaterialPack& pack : materialPacks ) {
+		for ( Material& material : pack.materials ) {
+			material.surfaceCommandBatchOffset = batchOffset;
+
+			const uint32_t cmdCount = material.drawCommands.size();
+			const uint32_t batchCount = cmdCount % SURFACE_COMMANDS_PER_BATCH == 0 ? cmdCount / SURFACE_COMMANDS_PER_BATCH
+																			   : cmdCount / SURFACE_COMMANDS_PER_BATCH + 1;
```

Comment:
That's a lot of space, you can park a whole train there! 😅️

Reply:
Heh, xhanged the spacing now.